model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_edge_diagnostics_for_dycore_and_update_vn.py

```diff
@@ -0,0 +1,678 @@
+        )
+        if (divdamp_order == combined_divdamp_order) | (divdamp_order == fourth_divdamp_order)
+        else broadcast(vpfloat("0.0"), (dims.EdgeDim, dims.KDim))
+    )
+
+    next_vn = (
+        where(
+            (start_edge_nudging_level_2 <= horz_idx < end_edge_local),
+            _add_temporal_tendencies_to_vn_by_interpolating_between_time_levels(
+                vn_nnow=current_vn,
+                ddt_vn_apc_ntl1=predictor_normal_wind_advective_tendency,
+                ddt_vn_apc_ntl2=corrector_normal_wind_advective_tendency,
+                ddt_vn_phy=normal_wind_tendency_due_to_physics_process,
+                z_theta_v_e=theta_v_at_edges_on_model_levels,
+                z_gradh_exner=horizontal_pressure_gradient,
+                dtime=dtime,
+                wgt_nnow_vel=wgt_nnow_vel,
+                wgt_nnew_vel=wgt_nnew_vel,
+                cpd=cpd,
+            ),
+            next_vn,
+        )
+        if itime_scheme == 4
+        else next_vn
+    )
+
+    next_vn = (
+        where(
+            (start_edge_nudging_level_2 <= horz_idx < end_edge_local),
+            _apply_2nd_order_divergence_damping(
+                z_graddiv_vn=horizontal_gradient_of_total_divergence,
+                vn=next_vn,
+                scal_divdamp_o2=second_order_divdamp_scaling_coeff,
+            ),
+            next_vn,
+        )
+        if (
+            (divdamp_order == combined_divdamp_order)
+            & (second_order_divdamp_scaling_coeff > 1.0e-6)
+        )
+        else next_vn
+    )
+
+    next_vn = (
+        where(
+            (start_edge_nudging_level_2 <= horz_idx < end_edge_local),
+            _apply_weighted_2nd_and_4th_order_divergence_damping(
+                scal_divdamp=fourth_order_divdamp_scaling_coeff,
+                bdy_divdamp=reduced_fourth_order_divdamp_coeff_at_nest_boundary,
+                nudgecoeff_e=nudgecoeff_e,
+                z_graddiv2_vn=squared_horizontal_gradient_of_total_divergence,
+                vn=next_vn,
+            ),
+            next_vn,
+        )
+        if (
+            (divdamp_order == combined_divdamp_order)
+            & (second_order_divdamp_factor <= (4.0 * fourth_order_divdamp_factor))
+            & limited_area
+        )
+        else next_vn
+    )
+
+    next_vn = (
+        where(
+            (start_edge_nudging_level_2 <= horz_idx < end_edge_local),
+            _apply_4th_order_divergence_damping(
+                scal_divdamp=fourth_order_divdamp_scaling_coeff,
+                z_graddiv2_vn=squared_horizontal_gradient_of_total_divergence,
+                vn=next_vn,
+            ),
+            next_vn,
+        )
+        if (
+            (divdamp_order == combined_divdamp_order)
+            & (second_order_divdamp_factor <= (4.0 * fourth_order_divdamp_factor))
+            & (not limited_area)
+        )
+        else next_vn
+    )
+
+    next_vn = (
+        where(
```

Comment:
I guess all domain `where` need to be substituted with `concat_where` now.

Reply:
I realized this just now and have the changes locally, will push soon. but thanks!